ios/TokenProvider.swift

```diff
@@ -6,21 +6,29 @@ enum TokenError: Error {
 
 class TokenProvider: ConnectionTokenProvider {
     static let shared = TokenProvider()
-    var completionCallback: ConnectionTokenCompletionBlock? = nil
-
+    var callbackMap: [String : ConnectionTokenCompletionBlock?] = [:]
     static var delegate: RCTEventEmitter? = nil
 
-    func setConnectionToken(token: String?, error: String?) {
+    func setConnectionToken(token: String?, error: String?, callbackId: String?) {
         let unwrappedToken = token ?? ""
         if (!unwrappedToken.isEmpty) {
-            self.completionCallback?(token, nil)
+            if let callbackId = callbackId {
```

Comment:
since this function doesn't do anything if the callback id is nil or it doesn't have a matching completion, I suggest using a single check at the top that logs a message and early returns if things aren't as they should be:

```
guard let callbackId,
     let completionCallback = self.callbackMap[callbackId] else { 
    print("⚠️ setConnectionToken requires the callbackId be set to the callbackId value provided to the tokenProviderHandler.")
    return
}
```

otherwise it would fail silently which would likely be tricky for people to debug.

and something similar on the Android side

Reply:
ok, let me improve it.